### DISCUSSION TITLE
Skip "SETTING UP TRUST" step

### DISCUSSION THREAD

**Discussion by AnatolManikalo** [SAP-archive/btp-setup-automator#394] [Q&A] [answered]
Hello colleagues,

We added the creation of **XSUAA** service "**apiaccess**" plan to our usecase and now there's an additional step of setting up trust to IdP. In our setup we use custom IdP and this step is not needed.

Is there a way to skip "SETTING UP TRUST" step if we have XSUAA service "apiaccess" plan in our usecase?

Thanks,
Anatol

**Comment by lechnerc77** [comment]
Hi @AnatolManikalo ,

I am afraid I do not fully understand the setup. Checking the [metadata](https://github.com/SAP-samples/btp-service-metadata/blob/main/v0/developer/xsuaa.json) of the XSUAA service and the service plan "apiaccess" I do not see any obligatory data that needs to be added to create a service instance.

Does the execution of the use case throw an error in this setup? 

Best,
Christian

**Comment by AnatolManikalo** [comment]
Hi @lechnerc77 

Thanks for your reply. 

After adding XSUAA service and the service plan "apiaccess" with servicekeys to usecase, during the setup of the subaccount, it starts fetching IAS tenants and asks to select one of the available. If I specify "iashost" in paramters.json it picks up what is mentioned there. Otherwise there's an interactive choice from the list of available IPDs.

![image](https://user-images.githubusercontent.com/32706007/211562139-7bf1f469-7b55-41c0-a260-4df217899dfc.png)

**Comment by AnatolManikalo** [comment]
[Archive.zip](https://github.com/SAP-samples/btp-setup-automator/files/10383061/Archive.zip)

Here's pramaters and usecase for reference

**Comment by AnatolManikalo** [comment]
I can see this usecase for trust creation has only one service which is initiating trust creation: https://github.com/SAP-samples/btp-setup-automator/blob/main/usecases/released/setup_ias.json

However, in some cases XSUAA service and the service plan "apiaccess" with servicekeys is needed without setting up trust.

**Comment by lechnerc77** [comment]
Thanks for the explanation. I have to check and will come back to you

**Comment by lechnerc77** [comment]
@AnatolManikalo as a result of our internal discussion I created the issue #398. The issue represents an enhancement request  to enable the setup you described in this discussion.

**Comment by lechnerc77** [comment, marked as answer]
@AnatolManikalo the enhancement is available on `dev` (#398). See PR #399 for details of the solution.
